SDL/SDLJoystick.cpp

```diff
@@ -11,167 +11,209 @@ extern "C" {
+	}
+}
 
+void SDLJoystick::setUpPlayer1()
+{
 	int numjoys = SDL_NumJoysticks();
-	SDL_JoystickEventState(SDL_ENABLE);
+	int joyIndex = -1;
 	for (int i = 0; i < numjoys; i++) {
-		joys.push_back(SDL_JoystickOpen(i));
-//		printf("Initialized joystick %d: %s",i,SDL_JoystickNameForIndex(i));
-		if(strstr(SDL_JoystickNameForIndex(i),"PLAYSTATION(R)3 Controller"))
-			g_Config.bPS3Controller = true;
+		if (joyIndex == -1 && SDL_IsGameController(i))
+		{
+			joyIndex = i;
+			break;
+		}
+	}
+	if (joyIndex != -1)
+	{
+		printf("Player 1 will be using joystick %d, %s\n", joyIndex, SDL_JoystickNameForIndex(joyIndex));
+		player1Controller = SDL_GameControllerOpen(joyIndex);
+		if (player1Controller && SDL_GameControllerGetAttached(player1Controller))
+		{
+			player1JoyInstanceID = SDL_JoystickInstanceID(SDL_GameControllerGetJoystick(player1Controller));
+			char *mapping = SDL_GameControllerMapping(player1Controller);
+			if (mapping == NULL)
+			{
+				printf("control pad %s is not in the game controller database, unable to set mappings!\n", SDL_JoystickNameForIndex(joyIndex));
+			}
+			else
+			{
+				printf("control pad mapping: %s\n", SDL_GameControllerMapping(player1Controller));
+			}
+		}
+		else
+		{
+			printf("Error! Could not open controller\n");
+		}
 	}
-
-        if (g_Config.bPS3Controller)
-                fillMappingPS3();
-        else
-                fillMapping();
-
-
 }
 
 SDLJoystick::~SDLJoystick(){
 	if (thread)
 	{
 		running = false;
+		if (player1Controller)
+		{
+			SDL_GameControllerClose(player1Controller);
+		}
 		SDL_Event evt;
 		evt.type = SDL_USEREVENT;
 		SDL_PushEvent(&evt);
 		SDL_WaitThread(thread,0);
 	}
-	for (SDL_Joystick *joy : joys)
-	{
-		SDL_JoystickClose(joy);
-	}
 }
 
 void SDLJoystick::startEventLoop(){
 	thread = SDL_CreateThread(SDLJoystickThreadWrapper, "joystick",static_cast<void *>(this));
 }
 
+keycode_t SDLJoystick::getKeycodeForButton(SDL_GameControllerButton button)
+{
+	switch (button)
+	{
+		case SDL_CONTROLLER_BUTTON_DPAD_UP: return NKCODE_DPAD_UP;
+		case SDL_CONTROLLER_BUTTON_DPAD_DOWN: return NKCODE_DPAD_DOWN;
+		case SDL_CONTROLLER_BUTTON_DPAD_LEFT: return NKCODE_DPAD_LEFT;
+		case SDL_CONTROLLER_BUTTON_DPAD_RIGHT: return NKCODE_DPAD_RIGHT;
+		case SDL_CONTROLLER_BUTTON_A: return NKCODE_BUTTON_2;
+		case SDL_CONTROLLER_BUTTON_B: return NKCODE_BUTTON_3;
+		case SDL_CONTROLLER_BUTTON_X: return NKCODE_BUTTON_4;
+		case SDL_CONTROLLER_BUTTON_Y: return NKCODE_BUTTON_1;
+		case SDL_CONTROLLER_BUTTON_RIGHTSHOULDER: return NKCODE_BUTTON_5;
+		case SDL_CONTROLLER_BUTTON_LEFTSHOULDER: return NKCODE_BUTTON_6;
+		case SDL_CONTROLLER_BUTTON_START: return NKCODE_BUTTON_10;
+		case SDL_CONTROLLER_BUTTON_BACK: return NKCODE_BUTTON_9; // select button
+		case SDL_CONTROLLER_BUTTON_GUIDE: return NKCODE_BACK; // pause menu
+	}
+	return NKCODE_UNKNOWN;
+}
+
 void SDLJoystick::ProcessInput(SDL_Event &event){
 	switch (event.type) {
-	case SDL_JOYAXISMOTION:
+		case SDL_CONTROLLERBUTTONDOWN:
```

Comment:
Understood. I'll sort this out together with the indentation as mentioned in previous comment.